File: android/app/src/main/java/com/ritualcircles/ui/views/HomeView.kt
package com.ritualcircles.ui.views

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.ritualcircles.ui.AppState
import kotlinx.coroutines.launch

@Composable
fun HomeView(state: AppState, onCircleDetails: () -> Unit) {
    val home = state.home
    val circle = home?.circle
    val session = home?.nextSession
    val my = home?.myAttendance
    val scope = rememberCoroutineScope()

    var working by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }

    Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
        if (circle != null) {
            Text("Your circle", style = MaterialTheme.typography.titleMedium)
            Text("${circle.ritualType} • ${circle.modality} • Weekly ${circle.recurringTime}", style = MaterialTheme.typography.bodySmall)
        }

        Text("Next session", style = MaterialTheme.typography.titleMedium)
        if (session != null) {
            Text(session.dateTime, style = MaterialTheme.typography.bodySmall)
            Text(session.locationOrLink, style = MaterialTheme.typography.bodySmall)

            Text(
                when (my?.status) {
                    "attending" -> "You’re coming."
                    "not_attending" -> "You haven’t committed yet."
                    else -> "Set your attendance."
                },
                style = MaterialTheme.typography.bodySmall
            )

            Button(
                enabled = !working,
                onClick = {
                    scope.launch {
                        working = true; error = null
                        try {
                            state.api().putAttendance(session.id, "attending")
                            state.refreshHome()
                        } catch (t: Throwable) { error = t.message }
                        working = false
                    }
                }
            ) { Text("I’m coming") }

            OutlinedButton(
                enabled = !working,
                onClick = {
                    scope.launch {
                        working = true; error = null
                        try {
                            state.api().putAttendance(session.id, "not_attending")
                            state.refreshHome()
                        } catch (t: Throwable) { error = t.message }
                        working = false
                    }
                }
            ) { Text("Not coming") }
        } else {
            Text("No upcoming session.", style = MaterialTheme.typography.bodySmall)
        }

        error?.let { Text(it, color = MaterialTheme.colorScheme.error) }

        Button(enabled = circle != null, onClick = onCircleDetails) { Text("Circle details") }
    }
}

File: ios/RitualCircles/RitualCircles/Views/HomeView.swift
import SwiftUI

struct HomeView: View {
    @EnvironmentObject private var state: AppState
    @State private var isWorking: Bool = false
    @State private var error: String?

    var body: some View {
        let home = state.home
        let circle = home?.circle
        let nextSession = home?.nextSession
        let myAttendance = home?.myAttendance

        VStack(spacing: 16) {
            if let circle {
                VStack(alignment: .leading, spacing: 8) {
                    Text("Your circle")
                        .font(.headline)
                    Text("\(circle.ritualType) • \(circle.modality.rawValue)")
                        .foregroundStyle(.secondary)
                    Text("Weekly: \(circle.recurringTime)")
                        .foregroundStyle(.secondary)
                }
                .frame(maxWidth: .infinity, alignment: .leading)
            }

            if let nextSession {
                VStack(alignment: .leading, spacing: 8) {
                    Text("Next session")
                        .font(.headline)
                    Text(nextSession.dateTime.formatted(date: .abbreviated, time: .shortened))
                    Text(nextSession.locationOrLink)
                        .foregroundStyle(.secondary)
                        .lineLimit(2)
                }
                .frame(maxWidth: .infinity, alignment: .leading)
            } else {
                Text("No upcoming session yet.")
                    .foregroundStyle(.secondary)
                    .frame(maxWidth: .infinity, alignment: .leading)
            }

            if let nextSession {
                // One clear action: set attendance (one tap)
                VStack(spacing: 10) {
                    if myAttendance?.status == .attending {
                        Text("You’re coming.")
                            .frame(maxWidth: .infinity, alignment: .leading)
                    } else if myAttendance?.status == .not_attending {
                        Text("You haven’t committed yet.")
                            .frame(maxWidth: .infinity, alignment: .leading)
                    } else {
                        Text("Set your attendance")
                            .frame(maxWidth: .infinity, alignment: .leading)
                    }

                    Button {
                        Task { await setAttendance(sessionId: nextSession.id, status: .attending) }
                    } label: {
                        Text(isWorking ? "Saving..." : "I’m coming")
                            .frame(maxWidth: .infinity)
                    }
                    .buttonStyle(.borderedProminent)
                    .disabled(isWorking)

                    Button {
                        Task { await setAttendance(sessionId: nextSession.id, status: .not_attending) }
                    } label: {
                        Text("Not coming")
                            .frame(maxWidth: .infinity)
                    }
                    .buttonStyle(.bordered)
                    .disabled(isWorking)
                }
            }

            if let error {
                Text(error).foregroundStyle(.red).font(.footnote)
                    .frame(maxWidth: .infinity, alignment: .leading)
            }

            if circle != nil {
                NavigationLink("Circle details") {
                    CircleDetailsView()
                }
                .frame(maxWidth: .infinity, alignment: .leading)
            }

            Spacer()
        }
        .padding()
        .refreshable {
            await state.refreshHome()
        }
    }

    private func setAttendance(sessionId: String, status: AttendanceStatus) async {
        isWorking = true
        error = nil
        defer { isWorking = false }

        do {
            _ = try await APIClient.shared.putAttendance(sessionId: sessionId, status: status)
            await state.refreshHome()
        } catch {
            self.error = error.localizedDescription
        }
    }
}

